### ISSUE TITLE
Proposal peer review

### ISSUE THREAD

**Issue by TylerHart740** [INFO-523-SU25/final-project-garcia-kyle#3]
The following is the peer review of the project proposal by [name of team completing peer review]. The team members that participated in this review are 

- [Tyler Hart] - @[TylerHart740]

- Describe the goal of the project.

The project is attempting to use kaggle data to build a predictive model to classify and predict network anomalies in normal network traffic. It is also aiming to answer the question of how supervised vs unsupervised machine learning models compare in predicting trends in the data. 

- Describe the data used or collected.

The data is a culmination of data from the kaggle competition it is a training and test data set. The training data set is 125,972 X 43 and the test data set is 22,453 X 43. The dataset was chosen because of the availability of the data as well as depth the single data set contains. There are 43 columns with the data seen in the table below: 

duration | int64 | Length (in seconds) of the connection.
protocol_type | object | Protocol used (e.g., tcp, udp, icmp).
service | object | Network service on the destination (e.g., http, telnet).
flag | object | Status flag of the connection.
src_bytes | int64 | Number of data bytes sent from source to destination.
dst_bytes | int64 | Number of data bytes sent from destination to source.
land | int64 | 1 if connection is from/to the same host/port; 0 otherwise.
wrong_fragment | int64 | Number of wrong fragments.
urgent | int64 | Number of urgent packets.
hot | int64 | Number of “hot” indicators.
num_failed_logins | int64 | Number of failed login attempts.
logged_in | int64 | 1 if successfully logged in; 0 otherwise.
num_compromised | int64 | Number of compromised conditions.
root_shell | int64 | 1 if root shell is obtained; 0 otherwise.
su_attempted | int64 | 1 if “su root” command attempted; 0 otherwise.
num_root | int64 | Number of “root” accesses.
num_file_creations | int64 | Number of file creation operations.
num_shells | int64 | Number of shell prompts invoked.
num_access_files | int64 | Number of accesses to control files.
num_outbound_cmds | int64 | Number of outbound commands (always 0 in KDD99).
is_host_login | int64 | 1 if login is to a host account; 0 otherwise.
is_guest_login | int64 | 1 if login is to a guest account; 0 otherwise.
count | int64 | Number of connections to the same host in the past 2 seconds.
srv_count | int64 | Number of connections to the same service in the past 2 seconds.
serror_rate | float64 | % of connections with SYN errors.
srv_serror_rate | float64 | % of connections to the same service with SYN errors.
rerror_rate | float64 | % of connections with REJ errors.
srv_rerror_rate | float64 | % of connections to the same service with REJ errors.
same_srv_rate | float64 | % of connections to the same service.
diff_srv_rate | float64 | % of connections to different services.
srv_diff_host_rate | float64 | % of connections to different hosts on the same service.
dst_host_count | int64 | Number of connections to the destination host.
dst_host_srv_count | int64 | Number of connections to the destination host and service.
dst_host_same_srv_rate | float64 | % of connections to the same service on the destination host.
dst_host_diff_srv_rate | float64 | % of connections to different services on the destination host.
dst_host_same_src_port_rate | float64 | % of connections from the same source port.
dst_host_srv_diff_host_rate | float64 | % of connections to the same service from different hosts.
dst_host_serror_rate | float64 | % of connections with SYN errors to the destination host.
dst_host_srv_serror_rate | float64 | % of connections with SYN errors to the destination service.
dst_host_rerror_rate | float64 | % of connections with REJ errors to the destination host.
dst_host_srv_rerror_rate | float64 | % of connections with REJ errors to the destination service.
attack | object | Label indicating the type of attack or “normal”.
level | int64 | Severity or confidence score of the attack (if available).

This was copied directly from the proposal.

- Describe how the research question will be answered, e.g. what approaches / methods will be used.

The question is going to be answered by creating a new column is_anamalous which will make up of all the non-normal network traffic. From there they will conduct data preparation and dimension reduction to help lessen the load off processing unneeded data. they will then apply parallel paths 1 path for supervise learning using methods like neural networks and support vector machines the team will get metrics to determine accuracy on the test set with scores of accuracies and precision like ROC AUC, f1, etc. The other path will aim to do the same analysis with unsupervised methods like K-Means and DBSCAN to group similar observations. Following this the team will compare and write up the information for other teams to see. 

- Is there anything that is unclear from the proposal?

It is generally well populated! Good work guys! 

- Provide constructive feedback on how the team might be able to improve their project.

My only feedback is that for a two man team the data seems pretty well put together and that there is not a lot of room to exercise some extra thinking. In this proposal you also did not mention anything new which is a requirement of the project. Just something to think on. 

- What aspect of this project are you most interested in and would like to see highlighted in the presentation.

I am hoping to see some cool visuals outside of the ones seen in the proposal. I am hoping that you guys can come up with a novel way of predicting weird network traffic as it could be helpful to people in the future. 

- Provide constructive feedback on any issues with file and/or code organization.

- (Optional) Any further comments or feedback?

**Comment by TylerHart740** [comment]
Provide constructive feedback on any issues with file and/or code organization.

The code looks good here and is well formatted! Good Job!

(Optional) Any further comments or feedback?

**Comment by YouKnowJoey** [comment]
This is great feedback. If our performance is sub-par, and we have enough time, we would like to feature engineer artifacts that improve our predictions.
